### ISSUE TITLE
Panic when searching with stop-words in quotes

### ISSUE THREAD

**Issue by Northn** [meilisearch/meilisearch#3521]
**Describe the bug**
When performing a simple POST request with any stop-word in quote, MeiliSearch returns HTTP Error 500 and it panics

**To Reproduce**
1. Create an index
2. Add any stop-word to it by corresponding method, for example, hello
3. Perform a POST search request with that stop-word in quotes, and add some word before/after these quotes, e.g.: "`some \"hello\"`"
4. See the error

**Expected behavior**
Search request has to be completed without any error

**Screenshots**
None

**Meilisearch version:**
1.0.1, but applies to 1.0.0 too

**Additional context**
Ubuntu 20.04, Debian, OS does not matter
amd64
Also there's no error if nothing else added before/after the quotes, for example, it won't panic if searching "`\"hello\"`" instead of "`some \"hello\"`"

**Comment by curquiza** [comment]
Hello @Northn 
Thanks for the report 🙏

I succeeded in reproducing

- add documents
- add stop-words `["the", "of"]`
- search with `"q": "prince \"of\""`
- get a panic
```bash
{
    "message": "task 38 panicked",
    "code": "internal",
    "type": "internal",
    "link": "https://docs.meilisearch.com/errors#internal"
}
```

**Comment by curquiza** [comment]
Added to v1.0.2 milestones
We will release a patch release asap 😇

**Comment by curquiza** [comment]
Hello again @Northn 

A small question. What was your original use case, I mean, having stop words in the `"`, so for a phrase search? How did you get this? 😊

**Comment by Northn** [comment]
> Hello again @Northn
> 
> A small question. What was your original use case, I mean, having stop words in the `"`, so for a phrase search? How did you get this? 😊

Well, that's really funny situation. We made an addon to integrate Xenforo with Meilisearch, and obviously, all content needs to be indexed. After full indexation it has to find and cache similar threads list. So, uhm, someone accidentally created a thread with stopword in brackets, we got tons of errors in logs and reported 😂😂

**Comment by curquiza** [comment]
Closed by #3531, will be present in v1.0.2 release
